### PR TITLE
asserts: add identity assertion type

### PR DESCRIPTION
**Pull request by emgee**

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by mvo5** [comment]
LGTM

**Comment by zyga** [comment]
+1

File: asserts/identity.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	return id.Header("account-id")
+}
+
+// DisplayName returns the human-friendly name for the identity.
+func (id *Identity) DisplayName() string {
+	return id.Header("display-name")
+}
+
+// Validation returns the authority's confidence in the identity.
+func (id *Identity) Validation() string {
```
Comment (niemeyer) [inline asserts/identity.go]: Can we please replace this by IsCertified() bool?

It won't be good to have arbitrary code doing string comparisons on this value. Better to do it only once here.

**Comment by niemeyer** [comment]
LGTM assuming the suggested API change.

**Comment by emgee** [comment]
I'm happy to make this change, but it will be the only (I think) method that does not match the name of the header field.

Maybe it would be better to change the header field to "certified", make it a false/true value, and call the corresponding method just Certified()?

**Comment by niemeyer** [comment]
These types are an API for the assertion, for us to have proper code around them. We shouldn't feel constrained to have only trivial getters in them.

The assertions payload, on the other hand, is a data carrying mechanism. For those, having a multi-valued field is much better than having a true/false one. Multi-valued fields carry more information in the same space (it's the validation that is certified), and allow more than two options.

**Comment by pedronis** [comment]
I think I agree with Gustavo here, I mostly at the start exported straight accessors to underlying fields because so many things were up in the air, not out of principle, the closer we get to finalize things and start using assertions in the client the more it makes sense to add helpers on the assertions that make sense from the POV of client usage, the service itself is mostly generic so there it won't matter either way

**Comment by emgee** [comment]
That's fine. I really just wanted to raise this for discussion as it's bit of a change from the current style of the API. I'll make the suggested change.

File: asserts/identity.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	_, err = checkMandatory(assert.headers, "display-name")
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = checkMandatory(assert.headers, "validation")
+	if err != nil {
+		return nil, err
+	}
+	certified := assert.headers["validation"] == identityValidationCertified
```
Comment (niemeyer) [inline asserts/identity.go]: It'd be okay to have compared that inside IsCertified itself, but that's fine too.

**Pull request merged**